adagrams/game.py

```diff
@@ -1,11 +1,71 @@
+import random
+
+
 def draw_letters():
-    pass
+    letters = (["A"] * 9 + ["B"] * 2 + ["C"] * 2 + ["D"] * 4 + ["E"] * 12
+               + ["F"] * 2 + ["G"] * 3 + ["H"] * 2 + ["I"] * 9 + ["J"]
+               + ["K"] + ["L"] * 4 + ["M"] * 2 + ["N"] * 6 + ["O"] * 8
+               + ["P"] * 2 + ["Q"] + ["R"] * 6 + ["S"] * 4 + ["T"] * 6
+               + ["U"] * 4 + ["V"] * 2 + ["W"] * 2 + ["X"] + ["Y"] * 2
+               + ["Z"])
+
+    hand = random.sample(letters, k=10)
+
+    return hand
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    word = word.upper()
+    for char in word:
+        if word.count(char) > letter_bank.count(char):
+            return False
+    return True
```

Comment:
This works fine! `count`, though must iterate over the `word` with each iteration of the for loop on line 19. Nothing wrong with a nested for loop, but here we are repeatedly iterating over the same characters more than once, so how could we avoid that? 

Maybe we can create a frequency map (aka a dictionary counting the frequency of items) of our `hand` and use the benefits of the dictionary's lookup time (O(1) constant) to reduce the nested loop (O(n^2)) to one loop (O(n)) Something like:
```python
for char in word:
    if char in hand_letter_freq and hand_letter_freq[char] > 0:
        hand_letter_freq[char] -= 1
    else:
        return False
return True
```